tests/ui/ui_tester.py

```diff
@@ -37,60 +39,197 @@ def add_headless(browser_options):
         # Test Firefox
         options = FirefoxOptions()
         add_headless(options)
-        self._driver = webdriver.Firefox(log_path='', firefox_options=options)
+        self.browser = webdriver.Firefox(log_path='', firefox_options=options)
```

Comment:
Why make this no longer a private variable?

Reply:
Child classes can use it too.